### PR TITLE
Don't block completion after end of type parameters in JSX elements

### LINKED ISSUE #47416: JSX attribute completions sometimes don't work when tag type arguments are specified
In the below example, completing after `GenericElement<number>` in `f1` doesn't work (no `xyz`), but in `f2`, it does. Note that this is not new and is unrelated to snippets (so my other PRs for JSX snippets don't change anything, we just found it in testing).

```tsx
declare const React: any;

namespace JSX {
    export interface IntrinsicElements {
        div: any;
    }
}

function GenericElement<T>(props: {xyz: T}) {
    return <></>
}

function fn1() {
    return <div>
        <GenericElement<number> />
    </div>
}

function fn2() {
    return <>
        <GenericElement<number> />
    </>
}

```

**Playground Link:** [Provided](https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEYD2A7AzgF3gJRFMGAXPFCgJ4DcAUFSlALYhoAO+CAUgMoAa8A3lXhD4IAB7MkMLAEsUGEDABmbeAEk5MWWmlgAohBCM5afoOHng0gG7FSlM0IC+VZ1UUBXFAWmp4AcRAUBR19Q0CMAB4AFQA+AApmGCRmNGI+UTIAL2IoxwBKU3M4DHcYFHgImIiAehiXGg8vDB9yxRQARjiCgSKQErKKyys683MIgKDNPQMjSJR3egAjBRj4WocK6qG610bvXzaAJi7C4WLS8sqNsYng6bC5CPmllbWR4RqdqiA)

### PR DESCRIPTION
**Pull request by jakebailey**
Fixes #47416

### CODE REVIEW

File: src/services/completions.ts [hunk 1]
```diff
@@ -2543,6 +2543,15 @@ namespace ts.Completions {
             }
 
             if (contextToken.kind === SyntaxKind.GreaterThanToken && contextToken.parent) {
+                // <Component<string> /**/ />
```
Comment (sandersn) [inline src/services/completions.ts:2546]: on its own, this fix makes sense to me. How does it relate to the next comment, which says there are two possibilities -- jsxtext inside a `<div>` and non-jsxtext inside a type-argument-having `<Component<T>`? It seems like the second case might be handled entirely by the new code, but I'm not sure. 

If it is, then the following check could be simplified.

**Review by sandersn — APPROVED** [review]
Looks good, although I have one question about the existing code.

File: src/services/completions.ts (hunk 1 above)
Comment (jakebailey) [reply src/services/completions.ts:2546]: I thought the same thing, but removing it breaks the test added in the original "don't do completions in JsxText" PR; I can't remember if I figured out why or not at this point, though, so can go double check.

Comment (jakebailey) [reply src/services/completions.ts:2546]: Seems like it's the comment that needs to be updated; the second listed possibility is no longer accurate as `location` in this case isn't `GreaterThanToken`, it's the entire element. I'll change the comment to reflect that it's only handling the first case.

**Pull request merged**